### DISCUSSION TITLE
Get admin template in frontend template

### DISCUSSION THREAD

**Discussion by FrDH** [solspace/craft-freeform#944] [Questions & Answers] [answered]
Is it possible to know the selected admin template in the frontend template?
I'm asking because we want to set up dynamic notifications but don't want to override the selected admin template (but without a specified template, the mail somehow isn't sent).

We currently have this:

````
{{ theForm.render({

    dynamicNotification: {
        recipients: block.adminRecipients|split(',')|map(o => o|trim),
        template: 'admin.twig'
    }

}) }}
````
This works, but it will always use the "admin.twig" template. But when I omit the "template" option, the mail isn't sent (is this intentionally or should it fallback to the template selected in the form?)

What I'm looking for is an optional override of the "template" option, or a way to get the selected admin template, like this:
````
dynamicNotification: {
    recipients: block.adminRecipients|split(',')|map(o => o|trim),
    template: theForm.adminnotifications.template
}
````
But obviously that doesn't work :)

**Comment by kjmartens** [comment, marked as answer]
Thanks for your feedback, @FrDH.

You _might_ be able to make something like this work by building a custom module (we have something that might get you most of the way there): https://docs.solspace.com/craft/freeform/v4/guides/building-a-custom-module/

Another option might be going about this the opposite way. You could use Twig inside the Admin Notification textarea to insert the recipients from the template. Some steps:

1. Don't use the `dynamicNotification` feature.
2. Create a new field to collect/store the email address(es) from `block.adminRecipients`.
3. Pass the `block.adminRecipients` data to Freeform's `overrideValues` parameter.

Your template code might look something like this:

``` twig
{{ craft.freeform.form("myForm", {
    overrideValues: {
        myNewEmailCollectorField: block.adminRecipients|split(',')|map(o => o|trim),
    }
}).render() }}
```

Your Admin Notify textarea might look something like this:

``` twig
{% if myNewEmailCollectorField -%}
{{ myNewEmailCollectorField.value }}
{%- else -%}
fallback@email.net 
{%- endif %}
```

https://docs.solspace.com/craft/freeform/v4/guides/passing-dynamic-data-to-forms/
https://docs.solspace.com/craft/freeform/v4/overview/email-notifications/#admin-notifications

**Comment by FrDH** [comment]
Sorry for the late reply, I'll be sure to check this out. Thanks for your time :)

**Comment by FrDH** [comment]
So the second approach works pretty well (don't know why it wouldn't), too bad the `myNewEmailCollectorField` needs to be added to the form but I think it's a workable solution.

Just a small note for future references:
I added `|join("\n")` to the value as the Admin Notify field instructs to separate multiple email addresses with a new line:

```
{{ craft.freeform.form("myForm", {
    overrideValues: {
        myNewEmailCollectorField: block.adminRecipients|split(',')|map(o => o|trim)|join("\n"),
    }
}).render() }}
```
I'm never sure on how different systems implement a new line (is it with "\n", "\r" or "\n\r"?), but the "\n" seems to work.

I also changed the Admin Notify field to this shorter notation:
```{{ myNewEmailCollectorField.value ??? "info@domain.com" }}```


Anyway, thanks a lot for taking the time to figure this out :)
